src/component/presentational/MyIconButton.jsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable react/jsx-tag-spacing,react/forbid-prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '@material-ui/core/IconButton/IconButton';
+import {
+  Clear, Delete, Done, Edit,
+} from '@material-ui/icons';
+import Tooltip from '@material-ui/core/Tooltip/Tooltip';
+import withStyles from '@material-ui/core/styles/withStyles';
+import { ButtonType, TooltipConst, UserListItemConst } from '../../constants';
+import { MyIconButtonStyles } from '../../styles';
+
+const MyIconButton = (props) => {
+  const {
+    classes, editId, user, onUserEdit, onUserRemoved, type, hover, hoverId,
+  } = props;
+  const showItem = (hover && hoverId === user.id) ? classes.show : classes.hide;
+
+  let icon;
+  let tooltip;
+
+  if (type === ButtonType.EDIT) {
```

Comment:
This if statement has been repeated in this file, in order to create the logic of two separated button into one component. The problem of this approach is if we need another type of button, we need to extend all of these if statements, and it'll get even more complicated and Not clean.
There are two other options, first, using inheritance and create a parent button class and then its children (delete, edit...). Second, the composition, which is recommended by React official docs:
https://reactjs.org/docs/composition-vs-inheritance.html